lib/spack/spack/test/spec_semantics.py

```diff
@@ -798,7 +798,7 @@ def test_spec_formatting(self):
         for named_str, prop in package_segments:
             expected = getattr(spec, prop, "")
             actual = spec.format(named_str)
-            assert str(expected) == actual
+            assert str(expected).strip() == actual
```

Comment:
Why is strip needed here?  If the new `__str__` adds spaces we should fix that.

Reply:
@tgamblin The new `__str__` removes whitespaces (here I am stripping the expected value not the actual one). This is due to the fact that I am stripping the return value of format after it is constructed since for anonymous specs:
1. `{arch=architecture}` starts with a space
2. `{compiler_flags}` starts and ends with a space

To be clearer on what the current implementation does in this PR:
```python
Spack version 0.16.1
Python 3.7.4, Linux x86_64
>>> import spack.spec
>>> s = spack.spec.Spec('hdf5 target=x86_64 cflags="-g" ^zlib ^mvapich2')
>>> str(s)
'hdf5 cflags="-g"  arch=linux-None-x86_64 ^mvapich2 ^zlib'
>>> s.format('{name}{arch=architecture}')
'hdf5 arch=linux-None-x86_64'
>>> s.format('{arch=architecture}')
'arch=linux-None-x86_64'
>>> s.format('{name}{compiler_flags}')
'hdf5 cflags="-g"'
>>> s.format('{compiler_flags}')
'cflags="-g"'
```
with the same input the implementation in `develop` would do instead (note the extra spaces):
```python
Spack version 0.16.1
Python 3.7.4, Linux x86_64
>>> import spack.spec
>>> s = spack.spec.Spec('hdf5 target=x86_64 cflags="-g" ^zlib ^mvapich2')
>>> str(s)
'hdf5 cflags="-g"  arch=linux-None-x86_64 ^mvapich2 ^zlib'
>>> s.format('{name}{arch=architecture}')
'hdf5 arch=linux-None-x86_64'
>>> s.format('{arch=architecture}')
' arch=linux-None-x86_64'
>>> s.format('{name}{compiler_flags}')
'hdf5 cflags="-g" '
>>> s.format('{compiler_flags}')
' cflags="-g" '
```
I was wondering if these spaces were there for a reason or not, as it seems removing them doesn't make any unit-test fail. Pinging also @becker33 since he last modified `Spec.format`.